### PR TITLE
Remove resource.WithBuiltinDetectors() which has not been maintained

### PR DESCRIPTION
**Pull request by zhyChesterCheung**
Resolved: #2026 
Remove resource.WithBuiltinDetectors() as well as the test function.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Removed the deprecated package `go.opentelemetry.io/otel/exporters/metric/prometheus`. (#2020)
 - Removed the deprecated package `go.opentelemetry.io/otel/exporters/trace/jaeger`. (#2020)
 - Removed the deprecated package `go.opentelemetry.io/otel/exporters/trace/zipkin`. (#2020)
+- Removed the resource.WithBuiltinDetectors() function which has not been maintained. (#2026)
```
Comment (jmacd) [inline CHANGELOG.md:28]:
```suggestion
- Removed the resource.WithBuiltinDetectors() function in favor of explicit WithFeature() options for every built-in detector. (#2026)
```

**Review by jmacd — APPROVED** [review]

**Comment by zhyChesterCheung** [comment]
Some link check has failed, because of the new experimental metrics v0.22.0. But this generally looks good, could anybody answer or help with this?

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 - Removed the deprecated package `go.opentelemetry.io/otel/exporters/trace/jaeger`. (#2020)
 - Removed the deprecated package `go.opentelemetry.io/otel/exporters/trace/zipkin`. (#2020)
+- Removed the resource.WithBuiltinDetectors() function in favor of explicit WithFeature() options for every built-in detector. (#2026)
```
Comment (MrAlias) [inline CHANGELOG.md:26]:
```suggestion
- Removed the `"go.opentelemetry.io/otel/sdk/resource".WithBuiltinDetectors` function.
  The explicit `With*` options for every built-in detector should be used instead. (#2026 #2097)
```

File: sdk/resource/builtin.go [hunk 3]
```diff
@@ -40,6 +40,7 @@ type (
 	// disable them.
 	host struct{}
 
+	// stringDetector is a Detector that provides information about the
```
Comment (MrAlias) [inline sdk/resource/builtin.go:43]: This is an incomplete sentence. Maybe something got cutoff?

File: sdk/resource/builtin.go [hunk 4]
```diff
@@ -78,7 +79,8 @@ func StringDetector(schemaURL string, k attribute.Key, f func() (string, error))
 	return stringDetector{schemaURL: schemaURL, K: k, F: f}
 }
 
-// Detect implements Detector.
+// Detect returns a *Resource that describes the string as a value
+// corresponding to k as well as the specific schemaURL.
```
Comment (MrAlias) [inline sdk/resource/builtin.go:83]: What does `k` refer to here?

File: sdk/resource/config.go [hunk 5]
```diff
@@ -92,3 +85,27 @@ type schemaURLOption string
 func (o schemaURLOption) apply(cfg *config) {
 	cfg.schemaURL = string(o)
 }
+
+// WithOS adds all the OS attributes to the configured Resource.
```
Comment (MrAlias) [inline sdk/resource/config.go:89]: If we are going to move these options to this file can we move all the `WithOS*` and `WithProcess*` options as well?

File: sdk/resource/resource_test.go [hunk 6]
```diff
@@ -373,22 +373,6 @@ func TestNew(t *testing.T) {
 				"A": "B",
 			},
 		},
-		{
-			name:   "Builtins",
```
Comment (MrAlias) [inline sdk/resource/resource_test.go:377]: Can we add tests for `WithOS` and `WithProcess` to replace this?

**Comment by MrAlias** [comment]
> Some link check has failed, because of the new experimental metrics v0.22.0. But this generally looks good, could anybody answer or help with this?

#2106 should resolve the markdown link issue. Updating the branch will pull in the fix.

File: sdk/resource/builtin.go (hunk 4 above)
Comment (zhyChesterCheung) [reply sdk/resource/builtin.go:83]: It stands for `attribute.Key` .

Comment (MrAlias) [reply sdk/resource/builtin.go:82]: Please update the comment here to make it clear.

File: sdk/resource/process_test.go [hunk 7]
```diff
@@ -107,15 +107,15 @@ func restoreAttributesProviders() {
...
-	t.Run("WithProcess", testWithProcess)
+	t.Run("WithPID", TestWithProcessPID)
+	t.Run("WithExecutableName", TestWithProcessExecutableName)
+	t.Run("WithExecutablePath", TestWithProcessExecutablePath)
+	t.Run("WithCommandArgs", TestWithProcessCommandArgs)
+	t.Run("WithOwner", TestWithProcessOwner)
+	t.Run("WithRuntimeName", TestWithProcessRuntimeName)
+	t.Run("WithRuntimeVersion", TestWithProcessRuntimeVersion)
+	t.Run("WithRuntimeDescription", TestWithProcessRuntimeDescription)
+	t.Run("WithProcess", TestWithProcess)
```
Comment (MrAlias) [inline sdk/resource/process_test.go:118]: This runs these tests twice in the CI system, right? Can we remove this and keep the exported functions.

Comment (zhyChesterCheung) [reply sdk/resource/process_test.go]: Already done as mentioned above, please review it again. Thanks

File: CHANGELOG.md (hunk 2 above)
Comment (zhyChesterCheung) [reply CHANGELOG.md]: Done~, thanks for checking this PR for one more time.

**Comment by zhyChesterCheung** [comment]
If everything looks good, please help to approve and merge it. Thanks a lot ~

**Review by Aneurysm9 — APPROVED** [review]

**Review by MadVikingGod — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Comment by zhyChesterCheung** [comment]
Hi, it's a great honor for me to apply for being a member of OpenTelemetry community, after finished reading the [membership guidelines doc](https://github.com/open-telemetry/community/blob/main/community-membership.md)

Here is the list of my contributions to the OpenTelemetry project:
- PRs merged
  - [open-telemetry/opentelemetry-go#2022](https://github.com/open-telemetry/opentelemetry-go/pull/2022)
  - [open-telemetry/opentelemetry-go#2097](https://github.com/open-telemetry/opentelemetry-go/pull/2097)
  - [open-telemetry/opentelemetry-go#1973](https://github.com/open-telemetry/opentelemetry-go/pull/1973)
- SIG projects I am involved with
  - https://github.com/open-telemetry/opentelemetry-go

Could anybody please be my sponsor supporting me to do that?
@MrAlias @jmacd @paivagustavo @Aneurysm9 @MadVikingGod 
Deeply appreciate~

**Pull request merged**